corehq/ex-submodules/dimagi/utils/django/email.py

```diff
@@ -17,10 +17,30 @@
 LARGE_FILE_SIZE_ERROR_CODES = [LARGE_FILE_SIZE_ERROR_CODE, LARGE_FILE_SIZE_ERROR_CODE_ICDS_TCL]
 
 
+def get_valid_recipients(recipients):
+    """
+    This filters out any emails that have reported hard bounces or complaints to
+    Amazon SES
+    :param recipients: list of recipient emails
+    :return: list of recipient emails not marked as bounced
+    """
+    from corehq.util.models import BouncedEmail
+    valid_recipients = []
+    for recipient in recipients:
+        if not BouncedEmail.objects.filter(email=recipient).exists():
```

Comment:
Is there any chance that somewhere in email-land along the bounce path the email address gets reformatted, causing our bounce detector to record the address with a different format and therefore causing this check to fail? This is probably not a major concern, but seems like a possible failure mode that would be hard to detect.

Reply:
any email that doesn't get parsed by `BouncedEmailManager` still remains in the `commcarehq-bounces@dimagi.com` inbox. my next step is to set up a monthly reminder to check that inbox if any unprocessed emails are left in there. I also am going to set up a process to delete all the "soft bounces" from the inbox (e.g. auto-responders)